examples/imaging_demo.py

```diff
@@ -62,57 +56,51 @@
     "2021-09-23T09:00:00",
     "2021-09-23T12:00:00",
 ]  # Set this range larger than the actual observation time
-energy_range = [28, 40]
+energy_range = [28, 40] * u.keV
 
 ###############################################################################
 # Create the meta pixel, A, B, C, D for the science and the background data
 
 meta_pixels_sci = create_meta_pixels(
-    cpd_sci, time_range=time_range_sci, energy_range=energy_range, phase_center=[0, 0] * u.arcsec, no_shadowing=True
+    cpd_sci, time_range=time_range_sci, energy_range=energy_range, flare_location=[0, 0] * u.arcsec, no_shadowing=True
 )
 
 meta_pixels_bkg = create_meta_pixels(
-    cpd_bkg, time_range=time_range_bkg, energy_range=energy_range, phase_center=[0, 0] * u.arcsec, no_shadowing=True
+    cpd_bkg, time_range=time_range_bkg, energy_range=energy_range, flare_location=[0, 0] * u.arcsec, no_shadowing=True
 )
 
 ###############################################################################
 # Perform background subtraction
 
 meta_pixels_bkg_subtracted = {
+    **meta_pixels_sci,
     "abcd_rate_kev_cm": meta_pixels_sci["abcd_rate_kev_cm"] - meta_pixels_bkg["abcd_rate_kev_cm"],
     "abcd_rate_error_kev_cm": np.sqrt(
         meta_pixels_sci["abcd_rate_error_kev_cm"] ** 2 + meta_pixels_bkg["abcd_rate_error_kev_cm"] ** 2
     ),
 }
 
 ###############################################################################
-# Create visibilites from the meta pixels
+# Create visibilities from the meta pixels
 
 vis = create_visibility(meta_pixels_bkg_subtracted)
 
 ###############################################################################
 # Calibrate the visibilities
 
-# Extra phase calibration not needed with these
-uv_data = get_uv_points_data()
-vis.u = uv_data["u"]
-vis.v = uv_data["v"]
-
 cal_vis = calibrate_visibility(vis)
 
 ###############################################################################
 # Selected detectors 10 to 7
 
 # order by sub-collimator e.g 10a, 10b, 10c, 9a, 9b, 9c ....
 isc_10_7 = [3, 20, 22, 16, 14, 32, 21, 26, 4, 24, 8, 28]
-idx = np.argwhere(np.isin(cal_vis.isc, isc_10_7)).ravel()
+idx = np.argwhere(np.isin(cal_vis.meta["isc"], isc_10_7)).ravel()
 
 ###############################################################################
-# Create an ``xrayvsion`` Visibilities object
+# Slice the visibilities to detectors 10 - 7
 
-stix_vis = Visibilities(
-    cal_vis.obsvis[idx], cal_vis.u[idx], cal_vis.v[idx], amplitude_uncertainty=cal_vis.amplitude_error[idx]
-)
+vis10_7 = cal_vis[idx]
```

Comment:
This should use `Visibilities.index_by_label`

Reply:
So the label are 10a, 10b, etc so was easier to directly slice here. Also doesn't seem to like using a list `vis.index_by_label(['10a', '10b'])` doesn't work but `vis.index_by_label('10a', '10b')` does I know why but is this the API we want?

Reply:
I've opened TCDSolar/xrayvision/issues/71 to track this.